File: src/pocketcloud/event/impl/template/TemplateEvent.php
<?php

namespace pocketcloud\event\impl\template;

use pocketcloud\event\Event;
use pocketcloud\template\Template;

abstract class TemplateEvent extends Event {

    public function __construct(private Template $template) {}

    public function getTemplate(): Template {
        return $this->template;
    }
}

File: cloud-shared/src/main/java/de/pocketcloud/shared/event/template/TemplateEvent.java
package de.pocketcloud.shared.event.template;

import de.pocketcloud.api.component.template.ITemplate;
import de.pocketcloud.api.event.Event;

public abstract class TemplateEvent implements Event {

    protected final ITemplate template;

    public TemplateEvent(ITemplate template) {
        this.template = template;
    }
}